sky/provision/lambda_cloud/instance.py

```diff
@@ -0,0 +1,271 @@
+        for instance_id, instance in instances.items():
+            if instance.get('status') == 'active':
+                ready_instance_cnt += 1
+        if ready_instance_cnt == config.count:
+            break
+
+        time.sleep(POLL_INTERVAL)
+
+    assert head_instance_id is not None, 'head_instance_id should not be None'
+    return common.ProvisionRecord(
+        provider_name='lambda',
+        cluster_name=cluster_name_on_cloud,
+        region=region,
+        zone=None,
+        head_instance_id=head_instance_id,
+        resumed_instance_ids=[],
+        created_instance_ids=created_instance_ids,
+    )
+
+
+def wait_instances(region: str, cluster_name_on_cloud: str,
+                   state: Optional[status_lib.ClusterStatus]) -> None:
+    del region, cluster_name_on_cloud, state  # Unused.
+
+
+def stop_instances(
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+    worker_only: bool = False,
+) -> None:
+    raise NotImplementedError(
+        'stop_instances is not supported for Lambda Cloud')
+
+
+def terminate_instances(
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+    worker_only: bool = False,
+) -> None:
+    """See sky/provision/__init__.py"""
+    del provider_config
+    lambda_client = _get_lambda_client()
+    instances = _filter_instances(cluster_name_on_cloud, None)
+    # TODO (kmushegi): terminate all instances together since
+    # remove_instances accepts multiple instance ids
+    for instance_id, instance in instances.items():
+        logger.debug(f'Terminating instance {instance_id}: {instance}')
+        if worker_only and not instance['name'].endswith('-worker'):
+            continue
+        try:
+            lambda_client.remove_instances(instance_id)
+            logger.info(f'Terminated instance {instance_id}')
+        except Exception as e:  # pylint: disable=broad-except
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError(
+                    f'Failed to terminate instance {instance_id}: '
+                    f'{common_utils.format_exception(e, use_bracket=False)}'
+                ) from e
+
+
+def get_cluster_info(
+    region: str,
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> common.ClusterInfo:
+    del region  # unused
+    running_instances = _filter_instances(cluster_name_on_cloud, ['active'])
+    instances: Dict[str, List[common.InstanceInfo]] = {}
+    head_instance_id = None
+    for instance_id, instance_info in running_instances.items():
+        instances[instance_id] = [
+            common.InstanceInfo(
+                instance_id=instance_id,
+                # TODO (kmushegi): check if this is correct;
+                # external ip is preferred
+                internal_ip='',
+                external_ip=instance_info['ip'],
+                ssh_port=22,
+                tags={},
```

Comment:
Why not include tag information here? I suppose there are tags attached to the Lambda VM?

https://github.com/skypilot-org/skypilot/blob/c7ee6a22c2fbb912bbe8f83b2931062e6b4e4490/sky/skylet/providers/lambda_cloud/node_provider.py#L101-L122

Reply:
Lambda's API does not support tagging afaik so these tags would be internal to Skypilot? happy to keep them in but looking at the runpod implementation, tags are left empty: https://github.com/skypilot-org/skypilot/blob/master/sky/provision/runpod/instance.py#L173

Reply:
Yeah that makes sense. LGTM